Tasks/SonarQubePostTest/SonarQubeBuildBreaker.ps1

```diff
@@ -14,10 +14,8 @@ function BreakBuildOnQualityGateFailure
             return;
         }
        
-        $analysisId = WaitForAnalysisToFinish
-        $qualityGateStatus = QueryQualityGateStatus $analysisId
+        $qualityGateStatus = GetOrFetchQualityGateStatus
```

Comment:
The previous code seemed like a better separation of the functionality - having a separate step at the higher level that explicitly performs the blocking operation makes it obvious where this is happening. The new code buries the blocking operation several method calls deeper making it harder to work out where and when it occurs. _GetOrFetchQualityGateStatus_ doesn't suggest that a blocking operation will be performed the first time it is called.